File: platform/app/schemas/llm_probe.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class LLMProbeResponse(BaseModel):
    """Normalized payload for quick LLM liveness checks."""

    ok: bool
    provider: str
    base_url: str
    requested_model: Optional[str] = None
    resolved_model: Optional[str] = None
    latency_ms: int
    input_tokens: int = 0
    output_tokens: int = 0
    total_tokens: int = 0
    error_code: Optional[str] = None
    error_message: Optional[str] = None
    checked_at: datetime


class LLMConfigResponse(BaseModel):
    """Current LLM connection configuration (API key masked)."""

    api_key_set: bool
    api_key_masked: str
    base_url: str
    model: str
    timeout: float
    role_model_map: dict[str, str]


class LLMConfigUpdateRequest(BaseModel):
    """Request to update LLM connection settings at runtime."""

    api_key: Optional[str] = None
    base_url: Optional[str] = None
    model: Optional[str] = None
    timeout: Optional[float] = None

File: web/src/types/llmProbe.ts
/**
 * Query parameters for LLM quick probe endpoint.
 */
export interface LLMProbeRequest {
  /** Optional model override for this probe request. */
  model?: string;
  /** Optional timeout in milliseconds; backend validates range. */
  timeout_ms?: number;
}

/**
 * Normalized response payload for LLM quick liveness checks.
 */
export interface LLMProbeResponse {
  /** Whether probe succeeded at model-channel level. */
  ok: boolean;
  /** LLM provider identifier. */
  provider: string;
  /** Provider base URL used by backend client. */
  base_url: string;
  /** Model requested by caller. */
  requested_model?: string | null;
  /** Model resolved by provider/backend runtime. */
  resolved_model?: string | null;
  /** End-to-end probe latency in milliseconds. */
  latency_ms: number;
  /** Prompt token usage for this probe call. */
  input_tokens: number;
  /** Completion token usage for this probe call. */
  output_tokens: number;
  /** Total token usage for this probe call. */
  total_tokens: number;
  /** Normalized probe error code when ok=false. */
  error_code?: string | null;
  /** Human-readable error message when ok=false. */
  error_message?: string | null;
  /** Probe execution timestamp in ISO string format. */
  checked_at: string;
}

export interface LLMConfigResponse {
  api_key_set: boolean;
  api_key_masked: string;
  base_url: string;
  model: string;
  timeout: number;
  role_model_map: Record<string, string>;
}

export interface LLMConfigUpdateRequest {
  api_key?: string;
  base_url?: string;
  model?: string;
  timeout?: number;
}
